### ISSUE TITLE
FancyWM unable to correct forced re-scaling from other full-screen apps that run on a different resolution from the desktop underneath it

### ISSUE THREAD

**Issue by incorrigible7** [FancyWM/fancywm#106]
**Describe the bug**
>Be me. 
>Enjoy FancyWM layout on side monitor with discord, Steam and Blizzard app, and DnB youtube playing. 
>Get talked into trying out 'F2P' Diablo Immortal. 
>Game launches in default 'wtf is this resolution' graphic settings. 
>Go blind. 
>Go back to desktop (minimize or close, same issue).
>Find that the mouse location and where FancyWM detects the mouse to be irreparably misaligned (until FancyWM restart). 
>Curse Blizzard for ruining FancyWM. 

**To Reproduce**
1. Install and run Diablo Immortal (with a resolution different from the desktop it is running on). Honestly, it might specifically be a Diablo Immortal issue, unless literally every other new game I've run defaults to my desktop resolution. 

**Expected behavior**
Not sure. Clean up after the mouse pointer mismatching mess that Blizzard created, I guess. For Diablo Immortal, to continue sucking. 

**Desktop (please complete the following information):**
 - OS: win10
 - Version 21H2 64-bit

**Additional context**
Would be just my luck the dev is actually a Diablo Immortal fan. Please don't revoke my license.

**Comment by veselink1** [comment]
Thanks for reporting. I'm unable to reproduce, for now, might be related to having a second monitor and changing resolutions. Hopefully, chasing this bug down will not involve installing Diablo Immortal, but I am prepared to go to that extent if necessary.

**Comment by incorrigible7** [comment]
> Hopefully, chasing this bug down will not involve installing Diablo Immortal, but I am prepared to go to that extent if necessary.

Oh god, what have I done.

**Comment by incorrigible7** [comment]
So I think this is specifically a Diablo Immortal issue, as I have not run into the same problem with other full-screen programs even when running on a non-native resolution. I also quit that abomination of a game very early on so I have no qualms if you decide not to explore that specific avenue whatsoever lol. The only other time I've experienced this issue is when my VR headset comes into play. My specific headset (HP Reverb G2), for whatever reason, adds itself as three displays that appear on desktop>right click>display settings, and also forcibly and irreversibly adjusts the positioning of my other monitors by aligning the tops of them, until the headset is no longer detected (goes to sleep or unplugged) and puts them back where they were. Fairly certain the other headsets I've tried (some iteration of the Oculus and one of the older Pimax's) do not add themselves as displays. I have three monitors with the ones on either side in reverse portrait and portrait mode with the tops of the three not physically aligned, if that helps. 

Upon recreating the problem to take these screenshots, I've discovered the problem only occurs on my right most monitor, labeled '1' in display properties, with different behavior during headset detection and after it's no longer detected. While the headset is detected, the indicator no longer appears when assuming the correct mouse position (just past the top edge of the bar), and then displays incorrectly much lower after the headset is no longer detected (screenshot below). My guess is it's displaying above the visible screen space when it's not visible with the headset detected. 

Headset not detected:
![display settings before](https://user-images.githubusercontent.com/108477484/180878460-b2501fb0-2b99-4697-acb9-24f8ad8dbf63.png)

Headset detected:
![display settings after](https://user-images.githubusercontent.com/108477484/180878572-3d6b482d-80b1-4096-b265-47cd010f38e1.png)

Headset no longer detected, monitor '1':
![position after headset no longer detected](https://user-images.githubusercontent.com/108477484/180880709-db5ea23d-509d-4ed3-b28d-a0e98dd80e5a.png)
